subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollector.java

```diff
@@ -1,25 +1,15 @@
 package org.gradle.trace.pid;
 
-import org.gradle.BuildAdapter;
-import org.gradle.BuildResult;
 import org.gradle.api.internal.GradleInternal;
 import org.gradle.internal.nativeintegration.ProcessEnvironment;
-import org.gradle.trace.stream.AsyncWriter;
+import org.gradle.util.GFileUtils;
 
 import java.io.File;
 
 @SuppressWarnings("unused")
 public class PidCollector {
     public static void collect(GradleInternal gradle, File outFile) {
         ProcessEnvironment processEnvironment = gradle.getServices().get(ProcessEnvironment.class);
-        AsyncWriter<Long> writer = new AsyncWriter<>(outFile, (l, w) -> w.print(l));
-        writer.append(processEnvironment.getPid());
-        writer.finished();
-        gradle.addBuildListener(new BuildAdapter(){
-            @Override
-            public void buildFinished(BuildResult result) {
-                writer.stop();
-            }
-        });
+        GFileUtils.writeFile(processEnvironment.getPid().toString(), outFile);
```

Comment:
Looks nicer. Any idea why we had the complex solution originally?

Reply:
It was meant to avoid hiting the disk at configuration time as explained in the commit message